src/App.tsx

```diff
@@ -17,18 +57,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+
+              <TodoFilter
+                setFilter={setCurrentFilter}
+                setSearch={setCurrentSearch}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+
+              <TodoList
+                todos={searchedTodos}
+                setShowModal={setShowModal}
+                setOpenedTodo={setOpenedTodo}
+                openedTodo={openedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {loading && <Loader />}
```

Comment:
The `Loader` component is being rendered twice: once inside the block on line 68 and again on line 81. This redundancy can lead to unnecessary rendering and should be avoided. Consider removing one of these instances to ensure the loader is only displayed once when loading.